frontend/extensions/helpers.lua

```diff
@@ -0,0 +1,275 @@
+local Font = require("ui/font")
+local ConfirmBox = require("ui/widget/confirmbox")
+local _ = require("gettext")
+local lfs = require("libs/libkoreader-lfs")
+
+-- Plugin for global methods
+-- Some modules don't use this, but communicate directly with each other by exchanging global variables by triggering events. For an example, search for variable AX_remaining_time.
+
+-- for widgets use :extend instead of :new ?:
+local Helpers = WidgetContainer:extend{
+
+	debug_once_used = false,
+
+	file_paths = {},
+
+	garbage = nil,
+}
+
+function Helpers:init(ui)
+	self.ui = ui
+end
+
+function Helpers:getHomeDir()
+
+	local homedir = G_reader_settings:readSetting("home_dir")
+
+	local current_ebook
+
+	if homedir == nil then
+		current_ebook = G_reader_settings:readSetting("lastfile")
+
+		if current_ebook ~= nil then
+			homedir = string.gsub(current_ebook, '/[- .,:;?!_A-Za-z0-9]+', '')
+		end
+    end
+
+	return homedir
+end
+
+function Helpers:getKoreaderDir()
+
+    local settings_dir = DataStorage:getSettingsDir()
+
+    return string.gsub(settings_dir, '/settings', '')
+end
+
+-- use timeout = nil for second argument when calling for no timeout:
+function Helpers:alertDebug(message, timeout, dismiss_callback)
+
+	if dismiss_callback == nil then
+		UIManager:show(InfoMessage:new { text = message, icon_file = 'resources/info-bug.png', timeout = timeout })
+
+	else
+		UIManager:show(InfoMessage:new { text = message, icon_file = 'resources/info-bug.png', timeout = timeout, dismiss_callback = dismiss_callback })
+	end
+end
+
+function Helpers:alertDebugOnce(message, timeout, dismiss_callback)
+
+	if not self.debug_once_used then
+
+		self:alertDebug(message, timeout, dismiss_callback)
+
+		self.debug_once_used = true
+	end
+end
+
+-- use timeout = nil for second argument when calling for no timeout:
+function Helpers:alertError(message, timeout, dismiss_callback)
+
+	if dismiss_callback == nil then
+		UIManager:show(InfoMessage:new { text = message, icon_file = 'resources/info-error.png', timeout = timeout })
+
+	else
+		UIManager:show(InfoMessage:new { text = message, icon_file = 'resources/info-error.png', timeout = timeout, dismiss_callback = dismiss_callback })
+	end
+end
+
+-- use timeout = nil for second argument when calling for no timeout:
+function Helpers:alertInfo(message, timeout, dismiss_callback)
+
+	if dismiss_callback == nil then
+		UIManager:show(InfoMessage:new { text = message, timeout = timeout })
+
+	else
+		UIManager:show(InfoMessage:new { text = message, timeout = timeout, dismiss_callback = dismiss_callback })
+	end
+end
+
+-- use timeout = nil for second argument when calling for no timeout:
+function Helpers:alertNoIcon(message, timeout)
+
+	UIManager:show(InfoMessage:new { text = message, show_icon = false, timeout = timeout })
+end
+
+function Helpers:shuffle(tbl)
+	for i = #tbl, 2, -1 do
+		local j = math.random(i)
+		tbl[i], tbl[j] = tbl[j], tbl[i]
+	end
+end
+
+function Helpers:split(str, pat)
```

Comment:
This seems to be used solely as a helper function for split_path? For split in general, see http://koreader.rocks/doc/modules/util.html#gsplit